Life insuranc Networking

WEB3-GForce opened the pull request:
@bford 

This pull request contains the draft version of the network protocol.

The documentation has not been updated yet; I will be working on creating better tests/documentation along with reorganizing the code. As I do so, comments regarding the general structure of the code would be much appreciated.

WEB3-GForce commented:
@bford 

I've finished the major refactoring, clean-up, and documentation for lifePolicy.go. It's ready for you to look over.

I've written unit tests along with a functional test that simulates a server, a client, and insurers via go channels. Promise creation, certification, and secret reconstruction all work over the channels.

policyMessage.go has not been cleaned or properly documented yet. However, most of the code is about defining messages for the networking code and creating the marshalling code.

WEB3-GForce commented:
@bford 

I'm just about finished with the changes for lifePolicy.go. It's in pretty good condition for a review.

WEB3-GForce commented:
@bford 
@davidiw 

I've finished with the cleanup for policyMessage.go. I've also implemented the full suite of marshalling methods for all messages.

The code is documented and tested and ready for review.

bford commented:
Hi William, what's the new file called 'et' in the prifi top-level dir?  Was that a mistaken 'git add'?

WEB3-GForce commented:
That is odd. I'm not sure where that came from. I removed it and pushed the
change. Please let me know if I fixed it.

On Mon, May 4, 2015 at 8:21 AM, Bryan Ford <notifications@github.com> wrote:

> Hi William, what's the new file called 'et' in the prifi top-level dir?
> Was that a mistaken 'git add'?
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_DeDiS_prifi_pull_21-23issuecomment-2D98691400&d=AwMCaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=GPxwT4fhhmVz-1RB6SMPpOGPGmZuqB1ZodgT3dSLWWI&m=EuW6X2EyhOX7ko1o8ylNgCupiZ0a5TiwRETHW5kcLa0&s=gaCyelphZ5cLm9seZwKcEzavtneqHiPtcyjoUSolfFY&e=>
> .
>



-- 
William Edward Bailey, III
